package.json

```diff
@@ -21,7 +21,7 @@
   "license": "MIT",
   "dependencies": {
     "bson": "^6.10.3",
-    "kareem": "2.6.3",
+    "kareem": "git@github.com:mongoosejs/kareem.git#vkarpov15/v3",
```

Comment:
As long as this here, this PR wont be merged, right? If so, i would recommend to mark this PR as WIP.

Reply:
I think it is safe to merge this into 9.0 branch so we can continue this work in a separate PR.